### ISSUE TITLE
Sort by Title?

### ISSUE THREAD

**Issue by rcodehub** [FreshRSS/FreshRSS#1629]
I'd like to be able to sort by title rather than date as I have some feeds that are better suited to be read like that.  I don't see any existing options to be able to do that as I suppose it goes against the nature of reading the most current news first.

I've been experimenting slightly at implementing such a feature and the closest I've been able to do has been to change the function sqlListWhere in EntryDOA.php from 

. 'ORDER BY e.id ' . $order
to 
. 'ORDER BY e.title ' . $order or a variation like . 'ORDER BY e.id ASC'

This works well for one page of news articles but it does not scroll and load any more articles, it only says the feed is done, when it's not.  Basically this change loads one page at a time in Title-sorted order.

I'm not a good enough php coder to track down and fully understand all the points that exist to load the feeds so I'm stuck here, but I'd like to be able to continue and get this to be a working option, perhaps by adding a new navigation menu option "sort by title / sort by date" but I fear that it would take quite a bit of work.

Any ideas or help on how I can get this working? Or could a more knowledgeable member or developer get this implemented?

Thanks in advance and I hope to see this as an official feature.

**Comment by Alkarex** [comment]
Hello @rizergash ,
Indeed, this is not a built-in feature, and it would not be easy to implement properly. Indeed, when new articles are retrieved, sorting by title would require inserting them in the middle of some pre-existing ones. This would break auto-refresh, paging, mark-all-as-read, and synchronisation with mobile clients.

I think a first step would be to clarify at which time you would like to sort by title, and for how many articles (newly received articles, current window, current unread articles, all articles...).

For instance, it would be relatively easy to sort by titles the articles during a refresh job, so new articles would be sorted relatively to another, but not merged with already existing articles. So ending up with something like: ABCDEF-ABCDEF-ABCDEF-ABCDEF...

Maybe you could give an example of feed + screenshot.

**Comment by rcodehub** [comment]
What I would personally like to see is the ability to sort any amount of feeds by title, at any time.  Whether it is only the new articles when "Show only unread" is selected, or all articles if "show only read" is selected.  If one choice were easier to implement than the other, the priority would be for unread items.

Focusing on just the unread messages for the moment, I don't know exactly how FreshRSS loads and caches the items to display, but my thoughts are along the lines of the new items are loaded into a table/array/whatever you'd call it in php, sort them by title, and then display them as usual. 

The change I experimented with only changed one set of loaded articles at a time.

An example would be something like this...

Original feed: https://i.imgur.com/hsNDCkN.png
Sorted feed using my code change: https://i.imgur.com/glQyxGp.png
Scrolling down did load another page of feeds, starting in sorted order -- but out of order with the entire group of new items .... https://i.imgur.com/pOKyPd6.png

So at what point could the entirety of new items be sorted... that's my question and seems like the easiest possible solution.  i.e. the very first call of "select all unread items" -> sort by title optionally

**Comment by rcodehub** [comment]
That one line I changed actually does work pretty well.  I noticed in the generateEntries function that listWhereRaw is called to create the initial list of feeds using $maxFeedEntries.  So in my settings I changed the max items to 500, and up to 500 items are sorted.

The code change does break some features though, at least to some extent.  For example the "Mark as read" navigation button and the "Mark all as read" at the bottom of the messages will mark a bunch of things as read but will leave some in the catagory, and subsequent pushes of the button will mark/remove only 1 item at a time.  I could probably live with that, although annoying.
